### PR TITLE
chore: update react@17 version

### PR DESCRIPTION
**Pull request by andresin87**
**What kind of change does this PR introduce?**

- [ ] bugfix
- [ ] feature
- [ ] refactoring / style
- [x] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**

**Does this PR introduce a breaking change?**

Tests ok 👍
Styleguidist already launched updating react@16 to react@17

**Other information**
https://reactjs.org/blog/2020/10/20/react-v17.html

image

### CODE REVIEW

**Review by oriolpuig — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -141,10 +141,10 @@
     "lint-staged": "^10.3.0",
     "markdownlint-cli": "^0.23.2",
     "prettier": "*",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
+    "react": "^16.13.1 || ^17.0.0",
```
Comment (rolandjitsu) [inline package.json:144]: We should probably use 17 going forward and upgrade the rest of the dev deps as well.

**Review by midudev — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (andresin87) [reply package.json:144]: I don't know what other extra dev-deps you want tu upgrade. Plz let me know.

Comment (rolandjitsu) [reply package.json:144]: You left a note that:
> Styleguidist already launched updating react@16 to react@17

so I suppose we should update that as well (`react-styleguidist`)? These are just dev dependencies anyway, so it won't really matter to end users.

Comment (rolandjitsu) [reply package.json:144]: There's also:
```
    "@testing-library/react": "^11.0.2",
    "@testing-library/react-hooks": "^3.4.1",
    ...

    "@types/react": "^16.9.49",
    "@types/react-dom": "^16.9.8",
```

Comment (andresin87) [reply package.json:144]: I mean, despite of upgrading react to v17, styleguideist is currently rendering and working fine.

Comment (andresin87) [reply package.json:144]: I mean, despite of upgrading react to v17, styleguideist is currently rendering and working fine.

Comment (rolandjitsu) [reply package.json:144]: Sure. In any case, I suggest using `^17.0.0` (not `^16.13.1 || ^17.0.0`) and update the @types packages as well and the testing lib if necessary. I'll have a subsequent PR to update the rest of the deps.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]
Do note that there's also a yarn.lock file, so you need to update that as well.

File: package.json (hunk 1 above)
Comment (andresin87) [reply package.json:144]: updating...

I'll send it on an our.

**Comment by andresin87** [comment]
Updated 😉

**Comment by rolandjitsu** [comment]
@andresin87 could you update the yarn.lock as well?

**Comment by andresin87** [comment]
done

**Review by rolandjitsu — APPROVED** [review]

**Comment by rolandjitsu** [comment]
Tests are passing, but coveralls is under maintenance, so not waiting for coverage reports as no code has changed.

**Pull request merged**